### PR TITLE
gbXML Construction Imports

### PR DESCRIPTION
**Pull request by jasondegraw**
Add two unit tests that indicated that the constructions are being imported from gbXML. The error reports were complicated by some naming issues of the materials, which may have cause issues. Addresses #2423 and the rest of #2841.

### CODE REVIEW

File: openstudiocore/src/gbxml/ReverseTranslator.cpp [hunk 1]
```diff
@@ -725,26 +732,95 @@ namespace gbxml {
...
+                  } else if (spaceSurfaceType == "InteriorFloor") {
+                    // Swap roles of space and adjacentSpace
+                    surface.setSpace(*adjacentSpace);
+                    auto temp = space;
+                    space = adjacentSpace;
+                    adjacentSpace = temp;
+                    figuredOut = true;
+                  }
+                }
+              } else if (!adjacentSpaceSurfaceType.isEmpty()) {
```
Comment (macumber) [inline openstudiocore/src/gbxml/ReverseTranslator.cpp]: @jasondegraw is this block ever going to be hit?  the first attribute is empty but the second is set?  if so, should this logic be reversed?  e.g. the current surface is a floor which is ok if the adjacent surface is a ceiling?

File: openstudiocore/src/gbxml/ReverseTranslator.cpp [hunk 2]
```diff
@@ -623,6 +627,9 @@ namespace gbxml {
         surface.setSurfaceType("Wall");
       }else if (surfaceType.contains("UndergroundWall")){
         surface.setSurfaceType("Wall");
+        surface.setOutsideBoundaryCondition("Ground");
+        surface.setSunExposure("NoSun");
+        surface.setWindExposure("NoWind");
```
Comment (macumber) [inline openstudiocore/src/gbxml/ReverseTranslator.cpp]: @jasondegraw should we set boundary conditions for all cases?  how about for the other underground surfaces 'UndergroundCeiling', 'UndergroundSlab', 'SlabOnGrade', 'RaisedFloor', 'ExposedFloor'?  It looks like explicit boundary conditions from the xml would override these defaults on line 655 which is correct

File: openstudiocore/src/gbxml/Test/ReverseTranslator_GTest.cpp [hunk 3]
```diff
@@ -93,6 +100,302 @@ TEST_F(gbXMLFixture, ReverseTranslator_ZNETH)
...
+
+  auto surfs = model->getModelObjects<Surface>();
+
+  OptionalSurface osurf = model->getModelObjectByName<Surface>("T-1-5-I-F-6");
+  ASSERT_TRUE(osurf);
+  auto points = osurf->vertices();
+  ASSERT_EQ(4, points.size());
+  EXPECT_EQ(0.9144, points[1].y());
+
+  //auto omat = model->getModelObjectByName<StandardOpaqueMaterial>("Concrete: 100 [mm]");
```
[the remark this answers is not in the document]
Comment (jasondegraw) [reply openstudiocore/src/gbxml/Test/ReverseTranslator_GTest.cpp]: @macumber Once someone can get the unit conversion stuff to work, then the "alternate units" test file would allow these to be uncommented (if I did the manual conversions correctly). Right now these will all fail because the none of the material properties get converted and SI is assumed. So it is waiting on more fixes.

File: openstudiocore/src/gbxml/ReverseTranslator.cpp (hunk 1 above)
Comment (jasondegraw) [reply openstudiocore/src/gbxml/ReverseTranslator.cpp]: @macumber I'm not sure why this is showing here as a diff, because it's in develop already. In any case, it could be hit if the surface type isn't specified for both spaces, and I think we've agreed we'd prefer to see that specified for both spaces. I think you're right, the logic needs to be reversed (due the space role swap if currentSurfaceType and adjacentSurfaceType agree).

File: openstudiocore/src/gbxml/ReverseTranslator.cpp [hunk 4]
```diff
@@ -725,26 +732,95 @@ namespace gbxml {
...
-          if (adjacentSpace){
-            // DLM: we have issues if interior ceilings/floors are mislabeled, override surface type for adjacent surfaces
-            // http://code.google.com/p/cbecc/issues/detail?id=471
+          if (adjacentSpace) {
+
             std::string currentSurfaceType = surface.surfaceType();
-            surface.assignDefaultSurfaceType();
-            if (currentSurfaceType != surface.surfaceType()){
-              LOG(Warn, "Changing surface type from '" << currentSurfaceType << "' to '" << surface.surfaceType() << "' for surface '" << surface.name().get() << "'");
+            if (currentSurfaceType == "RoofCeiling" || currentSurfaceType == "Floor") {
```
Comment (macumber) [inline openstudiocore/src/gbxml/ReverseTranslator.cpp]: @jasondegraw If the surfaceType attributes are set those should override the default surface type even if it is not a roof or a floor right?
